### DISCUSSION TITLE
Option to disable the search index via the config file

### DISCUSSION THREAD

**Discussion by alexanderloewe** [craftcms/cms#10142] [Ideas] [answered]
We have a medium size site with about 500 entries, but the search index has a size of 150 MB in the database.

At the moment we do not use any search function on the frontend, so i'd likt to disable the building of a search index into the db.

**Comment by brandonkelly** [comment, marked as answer]
You can prevent elements’ search keywords from getting indexed using the `craft\services\Elements::EVENT_BEFORE_UPDATE_SEARCH_INDEX` event. Add this code to your module:

```php
use craft\events\ElementEvent;
use craft\services\Elements;

Craft::$app->elements->on(
    Elements::EVENT_BEFORE_UPDATE_SEARCH_INDEX,
    function(ElementEvent $event) {
        $event->isValid = true;
        $event->handled = true;
    }
);
```

Note that this will also affect the search bar above element indexes within the control panel, which will likely frustrate authors, so not recommended.

Instead of disabling element keyword indexing altogether, you can prevent all custom fields from getting indexed, which will reduce the overall index size considerably. To do that, go to **Settings** → **Fields** and edit each field with a search icon (<img width="14" alt="Search icon" src="https://user-images.githubusercontent.com/47792/143027638-c6970400-a66c-46fc-83f0-d2e89c5baa82.png">). For each one, uncheck its “Use this field’s values as search keywords” setting. Also go into any Matrix, Super Table, and Neo fields and ensure that their sub-fields have that same setting disabled.

Once you’ve updated each of your fields, you can truncate your `searchindex` table, and then run the following command to re-index each of your entries without their custom fields’ keywords:

```sh
php craft resave/entries --update-search-index
```

**Reply by alexanderloewe** [reply]
Perfect, I will try that one.

Sure, the search index is used in the backend as well ;)

**Reply by alexanderloewe** [reply]
We have over 150 fields, that would be a bit of work. And maybe we will need the search index in the future.

**Reply by brandonkelly** [reply]
Alternatively you can do it like so:

1. Run the following command to rebuild your project config data:

   ```sh
   php craft project-config/rebuild
   ```

2. Replace any instances of `searchable: true` with `searchable: false` within your `config/project/` folder, using a text editor’s Find & Replace feature.

3. Run the following command:

   ```sh
   php craft project-config/apply
   ```

**Reply by alexanderloewe** [reply]
Good one, I don't had that one in mind 😂
